expo/tests/scheduler/test_add_event.py

```diff
@@ -1,100 +1,331 @@
+                     response.content)
+        nt.assert_in(klass.description,
+                     response.content)
+        nt.assert_in("1:00",
+                     response.content)
+        nt.assert_in(str(klass.teacher),
+                     response.content)
+
+    def test_good_user_minimal_post(self):
+        Conference.objects.all().delete()
+        Room.objects.all().delete()
+        context = ClassContext()
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", context.bid.eventitem_id])
+        response = self.client.post(
+            url,
+            data={'event-day': context.days[0].pk,
+                  'event-time': "12:00:00",
+                  'event-location': context.room.pk,
+                  'event-max_volunteer': 3,
+                  'event-title': 'New Title',
+                  'event-description': 'New Description',
+                  },
+            follow=True)
+
+        assert_redirects(response, reverse('event_schedule',
+                                           urlconf='scheduler.urls',
+                                           args=["Class"]))
+
+        nt.assert_not_in('<ul class="errorlist">', response.content)
+        nt.assert_in('Events Information', response.content)
+        sessions = Event.objects.filter(eventitem=context.bid, max_volunteer=3)
+        nt.assert_equal(len(sessions), 1)
+        session = sessions.first()
+        nt.assert_equal(session.starttime,
+                        datetime.combine(context.days[0].day,
+                                         time(12, 0, 0, tzinfo=pytz.utc)))
+        nt.assert_in('New Title', response.content)
+        nt.assert_in(str(context.room), response.content)
+        nt.assert_in('3', response.content)
+
+    def test_good_user_invalid_submit(self):
+        Conference.objects.all().delete()
+        Room.objects.all().delete()
+        context = ClassContext()
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", context.bid.eventitem_id])
+        response = self.client.post(
+            url,
+            data={'event-day': context.days[0].pk,
+                  'event-time': "12:00:00",
+                  'event-location': 'bad room',
+                  'event-max_volunteer': 3,
+                  'event-title': 'New Title',
+                  'event-description': 'New Description',
+                  })
+
+        nt.assert_equal(response.status_code, 200)
+        nt.assert_in('<input id="id_event-title" name="event-title" ' +
+                     'type="text" value="New Title" />',
+                     response.content)
+        nt.assert_in("New Description",
+                     response.content)
+        nt.assert_in('<input id="id_event-max_volunteer" min="0" ' +
```

Comment:
This looks more like it. This makes me want to write some testing utils functions to handle working with the html more cleanly, but I'm not going to push for that here.